### ISSUE TITLE
Wrong redirection happens after login when CMD_URL_PATH is specified

### ISSUE THREAD

**Issue by clvs7-gh** [codimd/server#273]
Under below conditions, wrong redirection will happen:

**Required config to reproduce:**
- `CMD_URL_PATH` is specified (example: codimd)
- Reverse proxied to CodiMD like below:
```
nginx (https://www.example.com/codimd/ , Docker container)
  |
   -- CodiMD (http://0.0.0.0:3000/ , Docker container)
```

**Steps to reproduce:**
1. User attempts to access restricted page, and see 'You are not allowed to access this page' error first.
2. Then, user sign in to CodiMD.
3. User will be redirected to '/(noteid)' and nginx error will be shown.

**Expected behavior:**
User should be redirected to '/(CMD_URL_PATH)/(noteid)'

**Actual behavior:**
User is redirected to '/(noteid)'

This behavior is occurred because errors.js (https://github.com/codimd/server/blob/8ce7b285636c02b4215c838640bb748947c8317f/lib/errors.js#L10) has 2 issues:

1. I think `serverUrl` is typo. Correct one is `serverURL`.
2. `req.originalUrl` doesn't have directory of `CMD_URL_PATH` because proxied request doesn't contain it.

**Comment by clvs7-gh** [comment]
I have patch for this issue, but sign rule ( https://github.com/codimd/server/blob/master/CONTRIBUTING.md#sign-your-work ) prevents me from sending PR.
I would not like to use real name for my personal reason, sorry. If project author allow me to use username or pseudonyms, I'd be happy to send PR.

**Comment by SISheogorath** [comment]
Let me put it this way: How can we check? It would put so much burden on us that it's not feasible. What we make sure is that the provided author is the person that also signs the commit off. When you don't use your real name there, we can't proof or disproof this (except you would use something like "this is not my real name")

I'm not encouraging you to use a fake name, bit I want to tell you that we have not the resources to check the identity of our contributors and we definitely value privacy.

**Comment by clvs7-gh** [comment]
@SISheogorath

Thank you for your reply, and I'm sorry for the inconvenience.

> Also if you don't loon for the fame another option would be to provide the patchfile and I or someone else in the community could apply it to the source tree.

So you mean it is ok that I send you patchfile?
Fame is not necessary for me. If it is ok, I'll send you patchfile. I can send it in any form (here, gist, email, etc).

**Comment by SISheogorath** [comment]
Just drop me an email (my profile contains an address) and I'll give it a go :)

**Comment by clvs7-gh** [comment]
@SISheogorath 
I've sent you email. Thank you for your kind assistance.

**Comment by nmaas87** [comment]
Any news on this?

**Comment by SISheogorath** [comment]
it is fixed in master, iirc not yet in the 2.0 branch: https://github.com/codimd/server/commit/d389f45818b39e95fb22d3e7d06f7ba061d6f34a

**Comment by davidmehren** [comment]
As the fix already landed in `master` and will be part of 1.6.1, I'm going to close this. Thanks again @clvs7-gh for your contribution!
